src/baldr/graphtile.cc

```diff
@@ -325,12 +325,13 @@ GraphId GraphTile::GetTileId(const std::string& fname) {
   //run backwards while you find an allowed char but stop if not 3 digits between slashes
   std::vector<int> digits;
   auto last = pos;
-  for(--pos; pos >= 0; --pos) {
+  for(--pos; pos < last; --pos) {
```

Comment:
the fix is here, if pos wraps around its no longer less than last and the loop ends